Service/AdditionalSection/RecordBuilder.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service\AdditionalSection;
+
+use Algolia\AlgoliaSearch\Api\RecordBuilder\RecordBuilderInterface;
+use Algolia\AlgoliaSearch\Service\AlgoliaConnector;
+use Magento\Framework\DataObject;
+use Magento\Framework\Event\ManagerInterface;
+
+class RecordBuilder implements RecordBuilderInterface
+{
+    public function __construct(
+        protected ManagerInterface $eventManager,
+    ){}
+
+    /**
+     * Builds a Section record
+     *
+     * @param DataObject $entity
+     * @return array
+     */
+    public function buildRecord(DataObject $entity): array
+    {
+        $record = [
+            AlgoliaConnector::ALGOLIA_API_OBJECT_ID => $entity->getData('value'),
+            'value'                                 => $entity->getData('value'),
```

Comment:
Is this necessary?

Reply:
I guess it could be not necessary (can't remember how it would behave without setting an objectID), I just ported the behaviour from `AdditionalSectionHelper`. Just wanted not to break anything :D